src/test/java/integration/BaseTest.java

```diff
@@ -75,4 +77,10 @@ public static CompletableFuture<Boolean> allFalse(List<TemplateFieldInfo> arr) {
         }
         return CompletableFuture.supplyAsync(() -> true);
     }
+
+    private static RepositoryApiClientImpl createTestRepoClient(String servicePrincipalKey, AccessKey accessKey, String testHeader) {
+        RepositoryApiClientImpl testRepoClient = RepositoryApiClientImpl.CreateFromAccessKey(servicePrincipalKey, accessKey);
+        testRepoClient.okBuilder.addInterceptor(chain -> chain.proceed(chain.request().newBuilder().addHeader(testHeader, "true").build()));
```

Comment:
Is there a way to add default headers to the Repository Client in an easier way, like for the js and c# clients? Can be added in a different PR if that is possible. This way you wouldn't need to make the okBuilder public

Reply:
from my understanding, the only way to add/set headers that will be passed through for every request for retrofit (i.e. authorization and loadtest) is to add them in the Oauth interceptor. Right now, the only alternative solutions I can think of involve creating additional objects specifically used for the tests

Reply:
oh wait, i think their might be a specific java package that will enable me to access private variables, looking into this a little closer......

Reply:
ok so 2 shorter alternative solutions that i am thinking of here that don't involve creating an additional test object

1. creating and using getters/setters in the repositoryApiClientimpl object 
2. using the java.lang.reflect package to get the private variable

Reply:
I feel like it would be a bit cleaner if there was a function in `RepositoryApiClientImpl` for setting default headers. Like it could be `RepositoryApiClientImpl.SetDefaultRequestHeaders(Dictionary headers)` and inside the function, it would set the headers on the private okBuilder

Reply:
ok added a `setDefaultHeaders` and a `setLoadTestHeader` function in `RepositoryApiClientImpl`